x-pack/legacy/plugins/apm/public/components/app/ServiceMap/index.tsx

```diff
@@ -37,19 +37,51 @@ ${theme.euiColorLightShade}`,
 
 export function ServiceMap({ serviceName }: ServiceMapProps) {
   const {
-    urlParams: { start, end }
+    urlParams: { start, end, environment },
+    uiFilters
   } = useUrlParams();
 
+  const uiFiltersOmitEnv = useMemo(
+    () => ({
+      ...uiFilters,
+      environment: undefined
+    }),
+    [uiFilters]
+  );
+
   const { data } = useFetcher(
     callApmApi => {
+      if (serviceName && start && end) {
+        return callApmApi({
+          pathname: '/api/apm/service-map/{serviceName}',
+          params: {
+            path: {
+              serviceName
+            },
+            query: {
+              start,
+              end,
+              environment,
+              uiFilters: JSON.stringify(uiFiltersOmitEnv)
```

Comment:
We normally don't send `environment` like a separate query params but just keep it in `uiFilters`. Can't we do the same here?